File: Easy/romanNums.cpp
class Solution {
public:
  int romanToInt(string s) {
    int I = 1, IV = 4, V = 5, IX = 9, X = 10, XL = 40, L = 50, XC = 90, C = 100, CD = 400, D = 500, CM = 900, M = 1000;
    int num = 0;

    for (int i = 0; i < s.length(); i++) {
      string z1;
      z1.push_back(s[i]);

      string z2 = "";
      if (i + 1 < s.length()) {
        z2.push_back(s[i + 1]);
      }

      string combined = z1 + z2;

      if (combined == "IV" || combined == "IX" || combined == "XL" || combined == "XC" || combined == "CD" || combined == "CM") {
        if (combined == "IV") num += IV;
        else if (combined == "IX") num += IX;
        else if (combined == "XL") num += XL;
        else if (combined == "XC") num += XC;
        else if (combined == "CD") num += CD;
        else if (combined == "CM") num += CM;
        i++;
      }
      else {
        if (z1 == "I") num += I;
        else if (z1 == "V") num += V;
        else if (z1 == "X") num += X;
        else if (z1 == "L") num += L;
        else if (z1 == "C") num += C;
        else if (z1 == "D") num += D;
        else if (z1 == "M") num += M;
      }
    }
    return num;
  }
};

File: Easy/romanNums.rs
// problem 13

impl Solution {
  pub fn roman_to_int(s: String) -> i32 {
    let chars: Vec<char> = s.chars().collect();
      let mut i = 0;
      let mut num = 0;

      while i < chars.len() {
        let current = chars[i];
        let next = if i + 1 < chars.len() { Some(chars[i + 1]) } else { None };

        match (current, next) {
          ('I', Some('V')) => { num += 4; i += 2; }
          ('I', Some('X')) => { num += 9; i += 2; }
          ('X', Some('L')) => { num += 40; i += 2; }
          ('X', Some('C')) => { num += 90; i += 2; }
          ('C', Some('D')) => { num += 400; i += 2; }
          ('C', Some('M')) => { num += 900; i += 2; }

          _ => {
            num += match current {
            'I' => 1,
            'V' => 5,
            'X' => 10,
            'L' => 50,
            'C' => 100,
            'D' => 500,
            'M' => 1000,
            _ => 0,
            };
          i += 1;
        }
      }
    }
    num
  }
}